src/certs_downloader/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+        let id_ca_subdir_arg = if args.len() >= 8 {
+            Some(args[7].clone())
+        } else {
+            None
+        };
+        let perm_ca_subdir_arg = if args.len() >= 9 {
+            Some(args[8].clone())
+        } else {
+            None
+        };
+        Ok(Config {
+            api_url: args[1].clone(),
+            username: args[2].clone(),
+            password: args[3].clone(),
+            nonce: args[4].clone(),
+            directory: args[5].clone(),
+            part_subdir: part_subdir_arg,
+            id_ca_subdir: id_ca_subdir_arg,
+            perm_ca_subdir: perm_ca_subdir_arg,
+        })
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct KeyPair {
+    private: String,
+    public: String,
+}
+
+pub fn download_certs(config: &Config) -> Result<(), Box<dyn std::error::Error>> {
+    let client = Client::builder().cookie_store(true).build().unwrap();
+    let login_url = format!("{}/login", config.api_url);
+    let credential = format!(
+        "{{\"username\":\"{}\", \"password\":\"{}\"}}",
+        config.username, config.password
+    );
+    let _auth_resp = client
+        .post(login_url)
+        .header(reqwest::header::CONTENT_TYPE, "application/json")
+        .body(credential)
+        .send();
+
+    let base_url = format!("{}/applications", config.api_url);
+    let id_ca_dir = format!(
+        "{}/{}",
+        config.directory,
+        if config.id_ca_subdir.is_some() {
+            config.id_ca_subdir.clone().unwrap()
+        } else {
+            String::from("")
+        }
+    );
+    let perm_ca_dir = format!(
+        "{}/{}",
+        config.directory,
+        if config.perm_ca_subdir.is_some() {
+            config.perm_ca_subdir.clone().unwrap()
+        } else {
+            String::from("")
+        }
+    );
+    let part_dir = format!(
+        "{}/{}",
+        config.directory,
+        if config.part_subdir.is_some() {
+            config.part_subdir.clone().unwrap()
+        } else {
+            String::from("")
+        }
+    );
+    download_cert(&client, &base_url, "identity_ca.pem", None, &id_ca_dir)?;
+    download_cert(&client, &base_url, "permissions_ca.pem", None, &perm_ca_dir)?;
+    download_cert(&client, &base_url, "governance.xml.p7s", None, &config.directory)?;
+    download_cert(&client, &base_url, "key_pair", Some(&config.nonce), &config.directory)?;
+    download_cert(&client, &base_url, "permissions.xml.p7s", Some(&config.nonce), &part_dir)?;
+
+    let kp_file = format!("{}/key_pair", config.directory);
+    let kp_str = fs::read_to_string(&kp_file)?;
+    let kp: KeyPair = serde_json::from_str(&kp_str)?;
+
+    let public_file = format!("{}/identity.pem", part_dir);
+    let private_file = format!("{}/identity_key.pem", part_dir);
+    fs::File::create(public_file)?.write_all(kp.public.as_bytes())?;
+    fs::File::create(private_file)?.write_all(kp.private.as_bytes())?;
+    fs::remove_file(&kp_file)?;
+    Ok(())
+}
+
+fn download_cert(
```

Comment:
```suggestion
fn download_file(
```

The file that's downloaded may not be a certificate